db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -516,8 +517,11 @@ public boolean isCoveringIndex(ConglomerateDescriptor cd) throws StandardExcepti
             return false;
 
         IndexRowGenerator irg=cd.getIndexDescriptor();
-        int[] baseCols=irg.baseColumnPositions();
 
+        if (irg.isOnExpression())
+            return false;
```

Comment:
For now, simply take all expression-based indexes as non-covering indexes. Doing so limits the scope of this change to a manageable size for code review. Treating an expression-based index as a covering index is DB-10236 and will have its own PR.